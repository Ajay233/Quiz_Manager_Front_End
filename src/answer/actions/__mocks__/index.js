export const getAnswers = jest.fn()

export const setCurrentAnswer = jest.fn()

export const deleteAnswer = jest.fn()

export const addAnswer = jest.fn(() => { return { type: "ADD_ANSWER", payload: {} } })

export const updateAnswer = jest.fn()
