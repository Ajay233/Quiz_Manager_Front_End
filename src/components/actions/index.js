export const setToTopButton = (show=false) => {
  return {
    type: "SHOW_TO_TOP_BUTTON",
    payload: show
  }
}

export const setUrl = (url="") => {
  return {
    type: "SET_URL",
    payload: url
  }
}

export const enableLightMode = () => {
  return {
    type: "SET_THEME",
    payload: false
  }
}

export const enableDarkMode = () => {
  return {
    type: "SET_THEME",
    payload: true
  }
}

export const setLoaderState = (show=false, message="", label="") => {
  return {
    type: "SET_LOADER_STATE",
    payload: {
      show: show,
      message: message,
      label: label
    }
  }
}
