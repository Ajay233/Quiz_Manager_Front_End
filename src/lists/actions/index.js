import { get } from '../../axiosRequests/requests'
import { sessionExpired } from '../../utils/session'

export const getCategories = (endpoint, jwt) => {
  return (dispatch) => {
    get(endpoint, jwt).then((response) => {
      dispatch({
        type: "SET_CATEGORIES",
        payload: response.data
      })
    }).catch((error) => {
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(dispatch);
      }
    })
  }
}
