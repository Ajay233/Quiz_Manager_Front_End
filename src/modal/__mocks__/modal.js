export const Modal = jest.fn(() => { return null })
