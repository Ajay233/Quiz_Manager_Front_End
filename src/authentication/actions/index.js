import { post } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import history from '../../history'

export const setUser = (endpoint, loginDetails) => {

  return (dispatch) => {

    post(endpoint, loginDetails).then((response) => {
      dispatch({
        type: "SET_USER_LOGGED_IN",
        payload: {
          id: response.data.user.id,
          forename: response.data.user.forename,
          surname: response.data.user.surname,
          email: response.data.user.email,
          permission: response.data.user.permission,
          verified: response.data.user.verified,
          jwt: response.data.jwt,
          loggedIn: true,
          response: response
        }
      })
      history.push("/")
      dispatch(setNotification(`Welcome back ${response.data.user.forename}`, "loginSuccess", true))
    }).catch((error) => {
      if(error.response.data === "NOT VERIFIED"){
        const verifyMsg = "Your email has not yet been veirified.  You will need to verify your email before you can log in"
        dispatch(setNotification(verifyMsg, "warning", true))
      } else {
        dispatch({
          type: "SET_USER_LOGGED_IN",
          payload: {
            error: error.response
          }
        });
        dispatch(setNotification("The username or password you entered was incorrect", "error", true));
      }
    })
  }
}

export const logOutUser = () => {
  return {
    type: "LOG_OUT_USER",
    payload: {
      forename: "",
      surname: "",
      email: "",
      permission: "",
      verified: "",
      jwt: "",
      loggedIn: false
    }
  }
}

export const setVerficationProcess = (endpoint, verificationDetails) => {

  return (dispatch) => {

    post(endpoint, verificationDetails).then((response) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "completed",
          token: "",
          error: {}
        }
      })
    }).catch((error) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "not verified",
          token: verificationDetails.token,
          error: error.response
        }
      })
    })
  }
}

export const setRedirect = (status) => {
  return {
    type: "SET_REDIRECT",
    payload: {
      status: status
    }
  }
}

export const logOut = () => {
  return {
    type: "RESET_APP"
  }
}

// export const getUsers = (endpoint, header) => {
//   return {
//     type: "GET_USER",
//     payload: {
//       userId
//   }
// }
