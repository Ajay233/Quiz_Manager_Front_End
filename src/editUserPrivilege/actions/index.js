import { getUsingParams } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const setUserResults = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("users/findByEmail", param, jwt).then((response) => {
      dispatch({
        type: "SET_USER_RESULTS",
        payload: response.data
      })
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        const msg = error.response.data;
        dispatch(setNotification(msg, "error", true));
        console.log(error.response)
      }
    });
  }
}

export const clearUserResults = () => {
  return {
    type: "CLEAR_USER_RESULTS",
    payload: {}
  }
}
