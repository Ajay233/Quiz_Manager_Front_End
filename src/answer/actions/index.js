import { getUsingParams, post, put, del } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { hideModal } from '../../modal/actions'
import { sessionExpired } from '../../utils/session'
import history from '../../history'

// create
export const addAnswer = (data, jwt) => {
  return (dispatch) => {
    return post("answer/create", [data], jwt).then((response) => {
      dispatch({
        type: "ADD_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer created", "success", true));
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        console.log(error.response);
        dispatch(setNotification("Error - unable to create answer", "error", true));
      }
    });
  }
}

// read
export const getAnswers = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("answer/findByQuestionId", param, jwt).then((response) => {
      dispatch({
        type: "SET_ANSWERS",
        payload: response.data
      });
    }).catch((error) => {
      dispatch(setNotification("Error retrieving answers", "error", true));
    });
  }
}

// update (there was a call to setCurrentAnswer but have removed this as it's not needed)
export const updateAnswer = (body, jwt) => {
  return (dispatch) => {
    return put("answer/update", [body], jwt).then((response) => {
      dispatch({
        type: "UPDATE_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer updated", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(setNotification("Error - unable to update answer", "error", true));
      }
    });
  }
}

// delete
export const deleteAnswer = (config, jwt) => {
  return (dispatch) => {
    return del("answer/delete", config, jwt).then((response) => {
      dispatch(hideModal())
      dispatch({
        type: "DELETE_ANSWER",
        payload: config.data[0]
      })
      dispatch(setNotification("Answer deleted", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(hideModal())
        dispatch(setNotification("Error - Unable to delete this answer", "error", true))
      }
    });
  }
}

export const setCurrentAnswer = (answer) => {
  return {
    type: "SET_CURRENT_ANSWER",
    payload: answer
  }
}
