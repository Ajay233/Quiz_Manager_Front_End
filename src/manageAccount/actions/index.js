import { put } from '../../axiosRequests/requests'
import { setUserData } from '../../authentication/actions'
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const updateUser = (data, jwt) => {
  const emailChangeMsg = "Profile information updated - Your email was changed so your access has been restriced until your email is verified"
  const errorMsg = "Error updating your profile data, please check the details you provided and try again";
  const emailTaken = "An account with that email already exists"
  return (dispatch) => {
    return put('users/update', data, jwt).then((response) => {
      if(response.data.jwt === undefined){
        dispatch(setUserData(response.data, jwt))
        dispatch(setNotification("Profile information updated", "success", true))
      } else {
        dispatch(setUserData(response.data.user, response.data.jwt))
        dispatch(setNotification(emailChangeMsg, "success", true))
      }
    }).catch((error) => {
      console.log(error)
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else if(error.response.data === emailTaken){
        dispatch(setNotification(emailTaken, "error", true))
      } else {
        dispatch(setNotification(errorMsg, "error", true))
      }
    });
  }
}
