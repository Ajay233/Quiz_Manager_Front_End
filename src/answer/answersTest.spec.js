import React from 'react'
import { Router } from 'react-router-dom'
import Answers from './answers'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { setCurrentAnswer, deleteAnswer } from './actions'
import { setNotification } from '../notifications/actions'
import { showModal } from '../modal/actions'
import history from '../history'


jest.mock("./actions")
jest.mock("../notifications/actions")
jest.mock("../modal/actions")

describe("", () => {

  let answers;

  beforeEach(() => {
    answers = [
      {
        id: 1,
        answerIndex: 1,
        description: "testDescription",
        correctAnswer: false
      },
      {
        id: 2,
        answerIndex: 2,
        description: "testDescription2",
        correctAnswer: true
      },
      {
        id: 3,
        answerIndex: 3,
        description: "testDescriptio3",
        correctAnswer: false
      }
    ]
  })

  it("should render a list of answers with options for ADMIN users", () => {

    const userData = {
      permission: "ADMIN",
      jwt: "jwt"
    }

    const component = renderer.create(
      <Router history={history}>
        <Answers
          answers={answers}
          userData={userData}
          setCurrentAnswer={setCurrentAnswer}
          setNotification={setNotification}
          deleteAnswer={deleteAnswer}
          showModal={showModal}
        />
      </Router>
    )

    expect(component).toMatchSnapshot()
  })

  it("should render a list of answers without options for READ-ONLY users", () => {

    const userData = {
      permission: "READ-ONLY",
      jwt: "jwt"
    }

    const component = renderer.create(
      <Router history={history}>
        <Answers
          answers={answers}
          userData={userData}
          setCurrentAnswer={setCurrentAnswer}
          setNotification={setNotification}
          deleteAnswer={deleteAnswer}
          showModal={showModal}
        />
      </Router>
    )

    expect(component).toMatchSnapshot()
  })
})
