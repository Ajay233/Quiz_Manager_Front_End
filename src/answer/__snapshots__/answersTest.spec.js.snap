// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of answers with options for ADMIN users 1`] = `
<div>
  <div>
    <div
      className="answerContainer"
    >
      <div
        className="ansNumber"
      >
        1
      </div>
      <div
        className="ansDecription"
      >
        testDescription
      </div>
      <div
        className="correctAns"
      >
        <i
          className="far fa-times-circle red"
        />
      </div>
      <div
        className="options"
      >
        <a
          className="deleteOption"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash-alt red"
          />
           Delete
        </a>
        <a
          className="edit"
          href="/editAnswer"
          onClick={[Function]}
        >
          <i
            className="fas fa-edit blue"
          />
           Edit
        </a>
      </div>
    </div>
  </div>
  <div>
    <div
      className="answerContainer"
    >
      <div
        className="ansNumber"
      >
        2
      </div>
      <div
        className="ansDecription"
      >
        testDescription2
      </div>
      <div
        className="correctAns"
      >
        <i
          className="far fa-check-circle green"
        />
      </div>
      <div
        className="options"
      >
        <a
          className="deleteOption"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash-alt red"
          />
           Delete
        </a>
        <a
          className="edit"
          href="/editAnswer"
          onClick={[Function]}
        >
          <i
            className="fas fa-edit blue"
          />
           Edit
        </a>
      </div>
    </div>
  </div>
  <div>
    <div
      className="answerContainer"
    >
      <div
        className="ansNumber"
      >
        3
      </div>
      <div
        className="ansDecription"
      >
        testDescriptio3
      </div>
      <div
        className="correctAns"
      >
        <i
          className="far fa-times-circle red"
        />
      </div>
      <div
        className="options"
      >
        <a
          className="deleteOption"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash-alt red"
          />
           Delete
        </a>
        <a
          className="edit"
          href="/editAnswer"
          onClick={[Function]}
        >
          <i
            className="fas fa-edit blue"
          />
           Edit
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render a list of answers without options for READ-ONLY users 1`] = `
<div>
  <div>
    <div
      className="answerContainer"
    >
      <div
        className="ansNumber"
      >
        1
      </div>
      <div
        className="ansDecription"
      >
        testDescription
      </div>
      <div
        className="correctAns"
      >
        <i
          className="far fa-times-circle red"
        />
      </div>
      <div
        className="options"
      />
    </div>
  </div>
  <div>
    <div
      className="answerContainer"
    >
      <div
        className="ansNumber"
      >
        2
      </div>
      <div
        className="ansDecription"
      >
        testDescription2
      </div>
      <div
        className="correctAns"
      >
        <i
          className="far fa-check-circle green"
        />
      </div>
      <div
        className="options"
      />
    </div>
  </div>
  <div>
    <div
      className="answerContainer"
    >
      <div
        className="ansNumber"
      >
        3
      </div>
      <div
        className="ansDecription"
      >
        testDescriptio3
      </div>
      <div
        className="correctAns"
      >
        <i
          className="far fa-times-circle red"
        />
      </div>
      <div
        className="options"
      />
    </div>
  </div>
</div>
`;
